plugins/processors/lookup/README.md

```diff
@@ -0,0 +1,140 @@
+# Lookup Processor Plugin
+
+The Lookup Processor allows to use one or more files containing a lookup-table
+for annotating incoming metrics. The lookup is _static_ as the files are only
+used on startup. The main use-case for this is to annotate metrics with
+additional tags e.g. dependent on their source. Multiple tags can be added
+depending on the lookup-table _files_.
+
+The lookup key can be generated using a Golang template with the ability to
+access the metric name via `{{.Name}}`, the tag values via `{{.Tag "mytag"}}`,
+with `mytag` being the tag-name and field-values via `{{.Field "myfield"}}`,
+with `myfield` being the field-name. Non-existing tags and field will result
+in an empty string or `nil` respectively. In case the key cannot be found, the
+metric is passed-trough unchanged. By default all matching tags are added and
+existing tag-values are overwritten.
+
+## Global configuration options <!-- @/docs/includes/plugin_config.md -->
+
+In addition to the plugin-specific configuration settings, plugins support
+additional global and plugin configuration settings. These settings are used to
+modify metrics, tags, and field or create aliases and configure ordering, etc.
+See the [CONFIGURATION.md][CONFIGURATION.md] for more details.
+
+[CONFIGURATION.md]: ../../../docs/CONFIGURATION.md#plugins
+
+## Configuration
+
+```toml @sample.conf
+# Lookup a key derived from metrics in a static file
+[[processors.lookup]]
+  ## List of files containing the lookup-table
+  files = ["path/to/lut.json", "path/to/another_lut.json"]
+
+  ## Format of the lookup file(s)
+  ## Available formats are:
+  ##    json               -- JSON file with 'key: {tag-key: tag-value, ...}' mapping
+  ##    csv_key_name_value -- CSV file with 'key,tag-key,tag-value,...,tag-key,tag-value' mapping
+  ##    csv_key_values     -- CSV file with a header and 'key,tag-value,...,tag-value' mapping
+  # format = "json"
+
+  ## Template for generating the lookup-key from the metric.
+  ## This is a Golang template (see https://pkg.go.dev/text/template) to
+  ## access the metric name (`{{.Name}}`), a tag value (`{{.Tag "name"}}`) or
+  ## a field value (`{{.Field "name"}}`).
+  key = '{{.Tag "host"}}'
+```
+
+## File formats
+
+The following descriptions assume `key`s to be unique identifiers used for
+matching the configured `key`. The `tag-name`/`tag-value` pairs are the tags
+added to a metric if the key matches.
+
+### `json` format
+
+In the `json` format, the input `files` must have the following format
+
+```json
+{
+  "keyA": {
+    "tag-name1": "tag-value1",
+    ...
+    "tag-nameN": "tag-valueN",
+  },
+  ...
+  "keyZ": {
+    "tag-name1": "tag-value1",
+    ...
+    "tag-nameM": "tag-valueM",
+  }
+}
+```
+
+Please note that only _strings_ are supported for all elements.
```

Comment:
I think the fact that strings are only supported + this is read at the start need to be highlighted at the top.